### PR TITLE
[EdgeDB] Media schema

### PR DESCRIPTION
**Pull request by andrewmurraydavid**
[Monday Task](https://seed-company-squad.monday.com/boards/5989610236/views/132288519/pulses/5799662499)

Created Media Schema for EdgeDB

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/6157904435) by [Unito](https://www.unito.io)

### CODE REVIEW

File: dbschema/media.esdl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  type Audio extending Media::TemporalMedia {
+    overloaded _type: Media::Type {
+      constraint expression on (__subject__ = Media::Type.Audio)
+    }
+  }
+}
+
+
+module Media {
+  scalar type Type extending enum<
```
Comment (CarsonF) [inline dbschema/media.esdl:23]: Let's not specify a type enum here. We can use the actual `type` names for that.

File: dbschema/media.esdl [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Image,
+    Video,
+    Audio
+  >;
+  
+  abstract type MediaBase extending default::Resource {
```
Comment (CarsonF) [inline dbschema/media.esdl:29]: Not a Resource
```suggestion
  abstract type MediaBase {
```

File: dbschema/media.esdl [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    _type: Type;
+
+    file: default::File;
```
Comment (CarsonF) [inline dbschema/media.esdl:32]:
```suggestion
    required file: default::File;
```

File: dbschema/media.esdl [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    
+    altText: str;
+    caption: str;
+    mimeType: str;
```
Comment (CarsonF) [inline dbschema/media.esdl:36]:
```suggestion
    required mimeType: str;
```

File: dbschema/media.esdl [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  type Dimensions {
+    required width: int16;
+    required height: int16;
+  }
+  
+  type VisualMedia extending MediaBase {
```
Comment (CarsonF) [inline dbschema/media.esdl:44]:
```suggestion
  abstract type VisualMedia extending MediaBase {
```

File: dbschema/media.esdl [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    dimensions: Dimensions;
```
Comment (CarsonF) [inline dbschema/media.esdl:45]:
```suggestion
    required dimensions: Dimensions;
```

File: dbschema/media.esdl [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+    overloaded _type: Type {
+      constraint expression on (__subject__ = Type.Image or __subject__ = Type.Video)
+    }
+  }
+  
+  type TemporalMedia extending MediaBase {
```
Comment (CarsonF) [inline dbschema/media.esdl:52]:
```suggestion
  abstract type TemporalMedia extending MediaBase {
```

File: dbschema/media.esdl [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    `duration`: int32;
```
Comment (CarsonF) [inline dbschema/media.esdl:53]:
```suggestion
    required `duration`: float32;
```

**Review by CarsonF — CHANGES_REQUESTED** [review]
This is a great start @andrewmurraydavid!

I think these FQNs make the most sense
```
default::Media
Media::Image
Media::Video
Media::Audio
```
Really (`default::`)`Media` is not that important either, but `Media::Media` didn't feel right.
What do you think?

File: dbschema/media.esdl (hunk 2 above)
Comment (andrewmurraydavid) [reply dbschema/media.esdl:29]: Ok, cool. Still trying understand what is and what isn't a resource. Thanks for the call out!

**Review by CarsonF — APPROVED** [review]
LGTM!

**Pull request merged**